src/Sarif/Core/FileData.cs

```diff
@@ -92,7 +92,7 @@ private static string EncodeFileContents(IFileSystem fileSystem, string filePath
 
             if (mimeType != SarifWriters.MimeType.Binary)
             {
-                fileContents = Encoding.UTF8.GetBytes(fileSystem.ReadAllText(filePath, inputFileEncoding));
+                fileContents = System.Text.Encoding.UTF8.GetBytes(fileSystem.ReadAllText(filePath, inputFileEncoding));
```

Comment:
There are three bugs here:
1. Now that there's no default, you can't get the file contents as text if you don't know the `inputFileEncoding`. In that case, the best you can to is populate `fileContents.binary` by reading the raw bytes.
2. On line 102, you always Base64-encode the resulting string. But -- and this is a change in the spec that you wouldn't be aware of -- the intention of `fileContents.text` is to be human-readable text, so the spec now says *not* to Base64 encode the contents of `fileContents.text.
3. If you don't have an `encoding` on the `file` object, fall back to `run.defaultFileEncoding`.

So the procedure is:
```
get the encoding from the file or from the run (they might both be absent).
read the bytes from the file
if the file is MIME binary OR you don't have an encoding OR the encoding isn't one that .NET supports
    base64 encode the bytes
   put the resulting string in the Binary property
else
   decode the bytes into a string using the encoding
   put the resulting string in the Text property
```

Reply:
How can I get the fileData.encoding when I'm in the process of creating the fileData in the first place?

Reply:
The caller of `FileData.Create` passes `encoding` as an argument. That gets passed to `EncodeFileContents` in the `inputFileEncoding` parameter. The only trick is that the caller now needs to pass `defaultFileEncoding` (which it should have gotten from the `Run` object) to the `Create` method as well.

Reply:
OK currently I don't see any callers providing an encoding param except one unit test. The only call where I see the opportunity to get the Run.DefaultFileEncoding is in SarifLogger.CaptureFile.

Reply:
So now we are returning a string from EncodeFileContents that the caller puts into either Text or Binary based on conditions within EncodeFileContents. Since the caller has no way of knowing those conditions, I was thinking of changing EncodeFileContents to non-static "SetFileContents" and creating the FileContent property in there. Thoughts?

Reply:
Or static EncodeFileContents could return a FileContent instance

Reply:
Definitely the second one...